projects/packages/forms/src/blocks/contact-form/components/jetpack-field-controls.js

```diff
@@ -163,105 +157,260 @@ const JetpackFieldControls = ( {
+						} );
+					} }
+					toolsPanelDropdownMenuProps={ toolsPanelDropdownMenuProps }
+				>
+					{ ( isChoicesBlock || blockStyle === 'button' ) && (
+						<>
+							<ToolsPanelItem
+								panelId={ clientId }
+								hasValue={ () => typeof attributes.buttonBorderWidth === 'number' }
+								label={ __( 'Button Border Width', 'jetpack-forms' ) }
+								onDeselect={ () =>
+									setAttributes( {
+										buttonBorderWidth: undefined,
+									} )
+								}
+							>
+								<RangeControl
+									label={ __( 'Button Border Width', 'jetpack-forms' ) }
+									value={ attributes.buttonBorderWidth }
+									initialPosition={ 1 }
+									onChange={ setNumberAttribute( 'buttonBorderWidth' ) }
+									min={ 0 }
+									max={ 100 }
+									__nextHasNoMarginBottom={ true }
+								/>
+							</ToolsPanelItem>
+							<ToolsPanelItem
+								panelId={ clientId }
+								hasValue={ () => typeof attributes.buttonBorderRadius === 'number' }
+								label={ __( 'Button Border Radius', 'jetpack-forms' ) }
+								onDeselect={ () =>
+									setAttributes( {
+										buttonBorderRadius: undefined,
+									} )
+								}
+							>
+								<RangeControl
+									label={ __( 'Button Border Radius', 'jetpack-forms' ) }
+									value={ attributes.buttonBorderRadius }
+									initialPosition={ 0 }
+									onChange={ setNumberAttribute( 'buttonBorderRadius' ) }
+									min={ 0 }
+									max={ 100 }
+									__nextHasNoMarginBottom={ true }
+								/>
+							</ToolsPanelItem>
+						</>
+					) }
+					{ ( ! isChoicesBlock || formStyle === FORM_STYLE.OUTLINED ) && (
+						<>
+							<ToolsPanelItem
+								panelId={ clientId }
+								hasValue={ () => typeof attributes.borderWidth === 'number' }
+								label={ __( 'Border Width', 'jetpack-forms' ) }
+								onDeselect={ () =>
+									setAttributes( {
+										borderWidth: undefined,
+									} )
+								}
+							>
+								<RangeControl
+									label={ __( 'Border Width', 'jetpack-forms' ) }
+									value={ attributes.borderWidth }
+									initialPosition={ 1 }
+									onChange={ setNumberAttribute( 'borderWidth' ) }
+									min={ 0 }
+									max={ 100 }
+									__nextHasNoMarginBottom={ true }
+								/>
+							</ToolsPanelItem>
+							<ToolsPanelItem
+								panelId={ clientId }
+								hasValue={ () => typeof attributes.borderRadius === 'number' }
+								label={ __( 'Border Radius', 'jetpack-forms' ) }
+								onDeselect={ () =>
+									setAttributes( {
+										borderRadius: undefined,
+									} )
+								}
+							>
+								<RangeControl
+									label={ __( 'Border Radius', 'jetpack-forms' ) }
+									value={ attributes.borderRadius }
+									initialPosition={ 0 }
+									onChange={ setNumberAttribute( 'borderRadius' ) }
+									min={ 0 }
+									max={ 100 }
+									__nextHasNoMarginBottom={ true }
+								/>
+							</ToolsPanelItem>
+						</>
+					) }
+				</ToolsPanel>
+				<ToolsPanel>
```

Comment:
Is there a functional purpose for using a `ToolsPanel` here?

I think we could do without the overhead that the `ToolsPanel` incurs if we don't need its management of reset functionality or display of controls.

Reply:
My main concern is the styling, and that it stays consistent with core's styles if they change with the same padding, spacing and border.

I don't think there's another component that can match that.

I think it's ok, and it'll probably only be a short term measure anyway.

Reply:
Ironically, the original `PanelBody` that you're changing the field settings from being within, has the same matching grid layout and spacing as the `ToolsPanel`.

The short term measure aspect probably hints at the need to hold off on this PR for a final decision on inner blocks.